app/models/hostgroup.rb

```diff
@@ -232,4 +235,8 @@ def validate_subnet_types
     errors.add(:subnet, _("must be of type Subnet::Ipv4.")) if self.subnet.present? && self.subnet.type != 'Subnet::Ipv4'
     errors.add(:subnet6, _("must be of type Subnet::Ipv6.")) if self.subnet6.present? && self.subnet6.type != 'Subnet::Ipv6'
   end
+
+  def suggest_default_pxe_loader
+    self.pxe_loader = self.try(:operatingsystem).try(:preferred_loader) unless self.pxe_loader.present?
```

Comment:
Yeah, I know why it fails, this fixes it:

    self.pxe_loader ||= self.try(:operatingsystem).try(:preferred_loader) if self.respond_to?(:pxe_loader)

This is `after_initialize` hook and it will fail if you load a row from database without this column. The method then does not exist. Unfortunately, the code ended up this way as the least intrusive way of suggesting a value based on operating system selected. We tried hard to do this in views and it was worse.